pkg/datapath/linux/node.go

```diff
@@ -702,6 +702,25 @@ func getNextHopIP(nodeIP net.IP, link netlink.Link) (nextHopIP net.IP, err error
 			copy(nextHopIP, route.Gw.To16())
 			break
 		}
+
+		// Select a gw for the specified link if there are multi paths to the nodeIP
+		// For example, the nextHop to the nodeIP 9.9.9.9 from eth0 is 10.0.1.2,
+		// from eth1 is 10.0.2.2 as shown bellow.
+		//
+		// 9.9.9.9 proto bird metric 32
+		//        nexthop via 10.0.1.2 dev eth0 weight 1
+		//        nexthop via 10.0.2.2 dev eth1 weight 1
+		//
+		// NOTE: We currently don't handle multiple next hops, so only one next hop
```

Comment:
I am wondering whether we might end up in a situation where we select a different "GW" than the one returned by bpf_fib_lookup in the datapath? In this case we might drop a packet due to failed neigh resolution.

Reply:
> The current neighbor discovery installs only one next hop per device, so this limitation already exists.

:+1: Thanks for the clarification. Maybe you could create a GH issue, so that we don't forget about it?

Reply:
Sure! I will create one.

Reply:
https://github.com/cilium/cilium/issues/29069